### PR TITLE
Update libnetwork with fixes for duplicate IP addresses

### PR DESCRIPTION
**Pull request by thaJeztah**
This updates libnetwork to https://github.com/docker/libnetwork/commit/8892d7537c67232591f1f3af60587e3e77e61d41 to bring in IPAM fixes for duplicate IP addresses.

Full diff: https://github.com/docker/libnetwork/compare/3aca383eb555510f3f17696f9505f7bfbd25f0e5... 8892d7537c67232591f1f3af60587e3e77e61d41

- IPAM tests (https://github.com/docker/libnetwork/pull/2104) (no changes in vendored files)
- Fix for Duplicate IP issues  (https://github.com/docker/libnetwork/pull/2104)

Signed-off-by: Sebastiaan van Stijn <github@gone.nl>

**- What I did**

**- How I did it**

**- How to verify it**

**- Description for the changelog**

```Markdown
* Fix IPAM assigning duplicate IP addresses []()
```

### CODE REVIEW

File: hack/dockerfile/install/proxy.installer [hunk 1]
```diff
@@ -3,7 +3,7 @@
 # LIBNETWORK_COMMIT is used to build the docker-userland-proxy binary. When
 # updating the binary version, consider updating github.com/docker/libnetwork
 # in vendor.conf accordingly
-LIBNETWORK_COMMIT=ed2130d117c11c542327b4d5216a5db36770bc65
+LIBNETWORK_COMMIT=2112cace8efd4c7b71daccbf54897871e205f293
```
Comment (thaJeztah) [inline hack/dockerfile/install/proxy.installer]: updating userland-proxy accordingly, but it was not in sync with the vendor https://github.com/docker/libnetwork/compare/ed2130d117c11c542327b4d5216a5db36770bc65... 2112cace8efd4c7b71daccbf54897871e205f293 (don't think this brings in any changes in the proxy)

File: vendor/github.com/docker/libnetwork/vendor.conf [hunk 2]
```diff
@@ -50,5 +50,6 @@ github.com/vishvananda/netns 604eaf189ee867d8c147fafc28def2394e878d25
 golang.org/x/crypto 558b6879de74bc843225cde5686419267ff707ca
 golang.org/x/net 7dcfb8076726a3fdd9353b6b8a1f1b6be6811bd6
 golang.org/x/sys 07c182904dbd53199946ba614a412c61d3c548f5
+github.com/golang/sync fd80eb99c8f653c847d294a001bdf2a3a6f768f5
```
Comment (thaJeztah) [inline vendor/github.com/docker/libnetwork/vendor.conf]: umph; this needs to be fixed in libnetwork: `github.com/golang/sync` is not the canonical import path. This is the same package as `golang.org/x/sync`

```
vndr github.com/golang/sync
2018/03/14 11:54:37 Collecting initial packages
2018/03/14 11:55:00 Download dependencies
2018/03/14 11:55:00 Failed to parse config: invalid config format: // FIXME this should be golang.org/x/sync, which is already vendored above
```

Comment (thaJeztah) [reply vendor/github.com/docker/libnetwork/vendor.conf]: Opened https://github.com/docker/libnetwork/pull/2111 to fix that

**Comment by thaJeztah** [comment]
made this WIP, pending https://github.com/docker/libnetwork/pull/2111 to be merged, after which this needs to be updated.

**Comment by thaJeztah** [comment]
Actually; it's only used in a test, which is not vendored in, so we should be ok here

**Comment by thaJeztah** [comment]
ping @fcrisciani @ddebroy @vdemeester PTAL

**Review by fcrisciani — APPROVED** [review]
LGTM

**Comment by thaJeztah** [comment]
CI failed on some flaky tests, so I bumped the sha to include https://github.com/docker/libnetwork/pull/2111 and retrigger CI

**Comment by thaJeztah** [comment]
Tests still flaky on the same problem;

```
16:02:22 --- FAIL: TestServiceWithPredefinedNetwork (31.84s)
16:02:22 	daemon.go:283: [d7b900e6d5c08] waiting for daemon to start
16:02:22 	daemon.go:315: [d7b900e6d5c08] daemon started
16:02:22 	service_test.go:55: timeout hit after 30s: task count at 1 waiting for 0
16:02:22 	daemon.go:273: [d7b900e6d5c08] exiting daemon
16:02:22 === RUN   TestServiceWithIngressNetwork
16:02:54 --- FAIL: TestServiceWithIngressNetwork (32.64s)
16:02:54 	daemon.go:283: [d28413a7a2a93] waiting for daemon to start
16:02:54 	daemon.go:315: [d28413a7a2a93] daemon started
16:02:54 	service_test.go:110: timeout hit after 30s: task count at 1 waiting for 0
16:02:54 	daemon.go:273: [d28413a7a2a93] exiting daemon
```

**Review by yongtang — APPROVED** [review]
LGTM

**Comment by thaJeztah** [comment]
something odd on the s390x. Should not be related to this PR though https://jenkins.dockerproject.org/job/Docker-PRs-s390x/8927/console;

/cc @tophj-ibm 

```
17:50:44 ----------------------------------------------------------------------
17:50:44 FAIL: docker_cli_run_unix_test.go:1405: DockerSuite.TestRunPIDsLimit
17:50:44 
17:50:44 docker_cli_run_unix_test.go:1409:
17:50:44     out, _ := dockerCmd(c, "run", "--name", "skittles", "--pids-limit", "4", "busybox", "cat", file)
17:50:44 /go/src/github.com/docker/docker/vendor/github.com/gotestyourself/gotestyourself/icmd/command.go:65:
17:50:44     t.Fatalf(err.Error() + "\n")
17:50:44 ... Error: 
17:50:44 Command:  /usr/local/cli/docker run --name skittles --pids-limit 4 busybox cat /sys/fs/cgroup/pids/pids.max
17:50:44 ExitCode: 125
17:50:44 Error:    exit status 125
17:50:44 Stdout:   
17:50:44 Stderr:   runtime/cgo: pthread_create failed: Resource temporarily unavailable
17:50:44 SIGABRT: abort
17:50:44 PC=0x1306ca2 m=2 sigcode=4294967290
17:50:44 
17:50:44 goroutine 0 [idle]:
17:50:44 
17:50:44 goroutine 1 [runnable, locked to thread]:
17:50:44 github.com/opencontainers/runc/vendor/github.com/seccomp/libseccomp-golang._Cfunc_seccomp_load(0x14ad68e0, 0x14ad68e0)
17:50:44 	github.com/opencontainers/runc/vendor/github.com/seccomp/libseccomp-golang/_obj/_cgo_gotypes.go:427 +0x46
17:50:44 github.com/opencontainers/runc/vendor/github.com/seccomp/libseccomp-golang.(*ScmpFilter).Load.func1(0x14ad68e0, 0x14017d0)
17:50:44 	/tmp/tmp.g8yqeMNbId/src/github.com/opencontainers/runc/vendor/github.com/seccomp/libseccomp-golang/seccomp.go:682 +0x54
17:50:44 github.com/opencontainers/runc/vendor/github.com/seccomp/libseccomp-golang.(*ScmpFilter).Load(0xc4200faba0, 0x0, 0x0)
17:50:44 	/tmp/tmp.g8yqeMNbId/src/github.com/opencontainers/runc/vendor/github.com/seccomp/libseccomp-golang/seccomp.go:682 +0x9a
17:50:44 github.com/opencontainers/runc/libcontainer/seccomp.InitSeccomp(0xc42004ee40, 0xc42000e0e0, 0x0)
17:50:44 	/tmp/tmp.g8yqeMNbId/src/github.com/opencontainers/runc/libcontainer/seccomp/seccomp_linux.go:77 +0x41a
17:50:44 github.com/opencontainers/runc/libcontainer.(*linuxStandardInit).Init(0xc4200fc9f0, 0x8, 0xc42000e0e0)
17:50:44 	/tmp/tmp.g8yqeMNbId/src/github.com/opencontainers/runc/libcontainer/standard_init_linux.go:136 +0x986
17:50:44 github.com/opencontainers/runc/libcontainer.(*LinuxFactory).StartInitialization(0xc420096120, 0x0, 0x0)
17:50:44 	/tmp/tmp.g8yqeMNbId/src/github.com/opencontainers/runc/libcontainer/factory_linux.go:321 +0x35e
17:50:44 main.glob..func6(0xc420088640, 0x0, 0x0)
17:50:44 	/tmp/tmp.g8yqeMNbId/src/github.com/opencontainers/runc/init.go:24 +0x5e
17:50:44 reflect.Value.call(0x13841a0, 0x14011a8, 0x13, 0x13ebf56, 0x4, 0xc420043800, 0x1, 0x1, 0xc420043788, 0x13e16e0, ...)
17:50:44 	/usr/local/go/src/reflect/value.go:434 +0x87c
17:50:44 reflect.Value.Call(0x13841a0, 0x14011a8, 0x13, 0xc420043800, 0x1, 0x1, 0x15f3280, 0x127fc96, 0x13ec06a)
17:50:44 	/usr/local/go/src/reflect/value.go:302 +0x90
17:50:44 github.com/opencontainers/runc/vendor/github.com/urfave/cli.HandleAction(0x13841a0, 0x14011a8, 0xc420088640, 0x0, 0x0)
17:50:44 	/tmp/tmp.g8yqeMNbId/src/github.com/opencontainers/runc/vendor/github.com/urfave/cli/app.go:487 +0x136
17:50:44 github.com/opencontainers/runc/vendor/github.com/urfave/cli.Command.Run(0x13ec08a, 0x4, 0x0, 0x0, 0x0, 0x0, 0x0, 0x13fddd4, 0x51, 0x0, ...)
17:50:44 	/tmp/tmp.g8yqeMNbId/src/github.com/opencontainers/runc/vendor/github.com/urfave/cli/command.go:191 +0x8d8
17:50:44 github.com/opencontainers/runc/vendor/github.com/urfave/cli.(*App).Run(0xc420001980, 0xc42000c060, 0x2, 0x2, 0x0, 0x0)
17:50:44 	/tmp/tmp.g8yqeMNbId/src/github.com/opencontainers/runc/vendor/github.com/urfave/cli/app.go:240 +0x4c0
17:50:44 main.main()
17:50:44 	/tmp/tmp.g8yqeMNbId/src/github.com/opencontainers/runc/main.go:147 +0x812
17:50:44 
17:50:44 goroutine 6 [syscall]:
17:50:44 os/signal.signal_recv(0x0)
17:50:44 	/usr/local/go/src/runtime/sigqueue.go:131 +0xf6
17:50:44 os/signal.loop()
17:50:44 	/usr/local/go/src/os/signal/signal_unix.go:22 +0x1e
17:50:44 created by os/signal.init.0
17:50:44 	/usr/local/go/src/os/signal/signal_unix.go:28 +0x3c
17:50:44 
17:50:44 r0   0x0	r1   0x0
17:50:44 r2   0x0	r3   0x3ffa901e618
17:50:44 r4   0x0	r5   0x8
17:50:44 r6   0xfffffffffffffff8	r7   0xc41fffff10
17:50:44 r8   0xf1	r9   0x3ffa00008c0
17:50:44 r10  0x3ffa901f910	r11  0x1627e38
17:50:44 r12  0x1627e28	r13  0x14472c0
17:50:44 r14  0x12caf3e	r15  0x3ffa901e578
17:50:44 pc   0x1306ca2	link 0x12caf3e
17:50:44 /usr/local/cli/docker: Error response from daemon: cannot start a stopped process: unknown.
17:50:44 time="2018-03-14T17:50:44Z" level=error msg="error waiting for container: context canceled" 
17:50:44 
17:50:44 
17:50:44 Failures:
17:50:44 ExitCode was 125 expected 0
17:50:44 Expected no error
17:50:44 
17:50:44 
17:50:49 
```

**Review by abhi — APPROVED** [review]
LGTM

**Comment by tiborvass** [comment]
LGTM

**Pull request merged**